app/controllers/manage_evaluators_controller.rb

```diff
@@ -0,0 +1,179 @@
+  end
+
+  def select_phase
+    params[:phase_id] ? @phases.find(params[:phase_id]) : @phases.first
+  end
+
+  def fetch_evaluator_invitations
+    @challenge.evaluator_invitations.where(phase: @phase)
+  end
+
+  def fetch_existing_evaluators
+    @challenge.evaluators.
+      joins(:challenge_phases_evaluators).
+      where(challenge_phases_evaluators: { phase: @phase }).
+      distinct
+  end
+
+  # Create action helpers
+  def process_evaluator_invitation(email)
+    existing_invitation = @challenge.evaluator_invitations.find_by(email:, phase: @phase)
+    user = User.find_by(email:)
+
+    if existing_invitation
+      resend_invitation(existing_invitation)
+    elsif user && valid_evaluator_role?(user)
+      add_user_as_evaluator(user)
+    else
+      create_new_invitation(email)
+    end
+  end
+
+  # prevent duplicate evaluator invitations
+  def resend_invitation(invitation)
+    invitation.update(last_invite_sent: Time.current) # only update last_invite_sent for now
+    {
+      success: true,
+      message: "An invitation to this challenge has already been sent to " \
+               "#{invitation.email}. Invitation has been resent."
+    }
+  end
+
+  def valid_evaluator_role?(user)
+    VALID_EVALUATOR_ROLES.include?(user.role)
+  end
+
+  def add_user_as_evaluator(user)
+    cpe = ChallengePhasesEvaluator.find_or_create_by(challenge: @challenge, phase: @phase, user:)
+    if cpe.persisted?
+      { success: true, message: "#{user.email} has been added as an evaluator for this phase." }
+    else
+      { success: false, message: "Failed to add #{user.email} as an evaluator." }
+    end
+  end
+
+  def create_new_invitation(email)
+    invitation = @challenge.evaluator_invitations.new(evaluator_invitation_params.merge(phase: @phase))
+    if invitation.save
+      { success: true, message: "Invitation sent to #{email} for this challenge phase." }
+    else
+      { success: false, message: invitation.errors.full_messages.join(", ") }
+    end
+  end
+
+  def handle_successful_creation(result)
+    redirect_to challenge_manage_evaluators_path(@challenge, phase_id: @phase.id),
+                notice: result[:message]
+  end
+
+  def handle_failed_creation
+    @evaluator_invitations = fetch_evaluator_invitations
+    @existing_evaluators = fetch_existing_evaluators
+    render :index
+  end
+
+  # Destroy action helpers
+  def process_evaluator_removal(evaluator_type, evaluator_id)
+    case evaluator_type
+    when 'user'
+      remove_user_evaluator(evaluator_id)
+    when 'invitation'
+      remove_evaluator_invitation(evaluator_id)
+    else
+      { success: false, message: 'Invalid evaluator type' }
+    end
+  end
+
+  def remove_user_evaluator(evaluator_id)
+    evaluator = @challenge.evaluators.find_by(id: evaluator_id)
+    return { success: false, message: 'Evaluator not found' } unless evaluator
+
+    cpe = ChallengePhasesEvaluator.find_by(challenge: @challenge, phase: @phase, user: evaluator)
```

Comment:
you can use find_by! to raise RecordNotFound if there is none
```suggestion
    cpe = ChallengePhasesEvaluator.find_by!(challenge: @challenge, phase: @phase, user: evaluator)
```